.devcontainer/devcontainer.json

```diff
@@ -0,0 +1,38 @@
+// For format details, see https://aka.ms/devcontainer.json. For config options, see the
+// README at: https://github.com/devcontainers/templates/tree/main/src/rust
+{
+    "name": "Rust",
+    // Or use a Dockerfile or Docker Compose file. More info: https://containers.dev/guide/dockerfile
+    "image": "mcr.microsoft.com/devcontainers/base:ubuntu",
+
+    // Allow access to host machine
+    "runArgs": ["--network=host"],
+
+    // Use 'mounts' to make the cargo cache persistent in a Docker Volume.
+    "mounts": [
+        {
+            "source": "devcontainer-cargo-cache-${devcontainerId}",
```

Comment:
So what does this translate to? Is it using Docker `cache` mounts underneath? I mean something like `--mount=type=cache,target=/usr/local/cargo`, or is it specific to dev containers.

Reply:
you get a volume which is attached to dev container just like any other volume

```
docker volume ls | grep devcontainer-cargo-cache
local     devcontainer-cargo-cache-1u4p07avg375gi7o172o2cfupr5iafgjj9t7hvog8dscsv7co5df
```
since it is container specific, running the same container would give you same volume and helps avoiding repetitive whole world of internet download.

Reply:
Ok, so it's manually maintaining what `--mount=type=cache,...` gives you.

Do you know if it has a cached `target` directory as well for the output of the build? I'm just curious.

Reply:
I didn't try but I think you can as per spec https://github.com/devcontainers/spec/blob/main/schemas/devContainer.base.schema.json#L711C8-L711C12

you can bring your own arbitrary location if this is what you mean.